test/Kestrel.FunctionalTests/HttpProtocolSelectionTests.cs

```diff
@@ -16,8 +16,18 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
 {
-    public class HttpProtocolSelectionTests
+    public class HttpProtocolSelectionTests : IDisposable
     {
+        public HttpProtocolSelectionTests()
+        {
+            AppContext.SetSwitch(ListenOptions.Http2ExperimentSwitch, isEnabled: true);
+        }
+
+        public void Dispose()
+        {
+            AppContext.SetSwitch(ListenOptions.Http2ExperimentSwitch, isEnabled: false);
```

Comment:
This makes it impossible to add another functional test class for HTTP/2 while keeping the experimental flag. Can we create an internal flag like we did for the other switches so that we can test without using the switch itself?

Reply:
What if we turn the switch on in tests and don't turn it off?

Reply:
It would be nice to test the switch functionality by just setting the internal flag without risking the failure of other tests.

Reply:
*shrug* The switch is pretty trivial, an automated test does not seem necessary. We should opt the test project into HTTP/2 support, there's not much risk for cross contamination as you still need to make a deliberate setting in each test to enable it.

Reply:
This is dev, not a patch branch. I think we should have some test for the switch. If we want to have this test project opt-in to the switch, and then have another test project verify it's disabled by default that's also OK I guess.